app/javascript/controllers/flatpickr_controller.js

```diff
@@ -2,28 +2,70 @@ import { Controller } from "@hotwired/stimulus"
 import flatpickr from "flatpickr"
 
 const invalidClass = "is-invalid"
-const timeValidator = /^[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}$/
+const timeValidator = /^[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9][0-9]?:[0-9]{2}:[0-9]{2} (AM|PM)( [A-Z]+)?$/
```

Comment:
Added timezones and AM/PM to this input.  Timezone disappears on focus.  (You kind of have to try it to see).